sample/src/main/java/com/szugyi/circlemenusample/WheelActivity.java

```diff
@@ -0,0 +1,177 @@
+package com.szugyi.circlemenusample;
+
+import android.animation.ObjectAnimator;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.view.animation.DecelerateInterpolator;
+import android.widget.FrameLayout;
+import android.widget.ImageView;
+
+import com.szugyi.circlemenu.view.WheelLayout;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by KhaledLela on 9/30/17.
+ */
+
+public class WheelActivity extends AppCompatActivity{
+    private WheelLayout mcWheel;
+    private WheelLayout scWheel;
+    private final int ITEM_LIMIT = 20;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_wheel);
+        final View wheelView = findViewById(R.id.wheel_layout);
+        mcWheel = (WheelLayout) findViewById(R.id.mc_wheel);
+        scWheel = (WheelLayout) findViewById(R.id.sc_wheel);
+
+        wheelView.post(new Runnable() {
+            @Override
+            public void run() {
+                final int offset = getResources().getDimensionPixelSize(R.dimen.spin_pointer_width)/2;
+                int width = mcWheel.getWidth() - offset;
+
+                int height = mcWheel.getHeight();
+                int mSize = Math.min(width,height);
+
+                ((FrameLayout.LayoutParams)mcWheel.getLayoutParams()).width = mSize;
+                ((FrameLayout.LayoutParams)mcWheel.getLayoutParams()).height = mSize;
+
+                ((FrameLayout.LayoutParams)scWheel.getLayoutParams()).width = mSize;
+                ((FrameLayout.LayoutParams)scWheel.getLayoutParams()).height = mSize;
+            }
+        });
+        initWheel();
+        loadMC();
+        loadSCByMC(mcWheel.getSelectedWheelItem());
+    }
+
+    private void initWheel() {
+        mcWheel.setBgImageView((ImageView) findViewById(R.id.mc_bg));
+        scWheel.setBgImageView((ImageView) findViewById(R.id.sc_bg));
+        final ImageView sprLeft = (ImageView) findViewById(R.id.spr_left);
+        final ImageView sprRight = (ImageView) findViewById(R.id.spr_right);
+        scWheel.setOnRotateListener(new WheelLayout.OnRotateListener() {
+            @Override
+            public void onRotationFinished(View view) {
+                sprRight.setRotation(30);
+            }
+
+            private ObjectAnimator animator;
+            @Override
+            public void onStopAnimation() {
+                if (animator != null && animator.isRunning()) {
+                    animator.cancel();
+                    animator = null;
+                }
+            }
+
+            @Override
+            public void onRotate(float angle) {
+                final float rotation = sprRight.getRotation() - angle;
+                if (rotation > 83 || rotation <-23)
```

Comment:
what are these magic numbers?

Reply:
Limit for spin pointer rotation for sampling.